### PR TITLE
Remove Lowercase Transformation for Case-Sensitive URLs

### PR DESCRIPTION
**Pull request by mujtabachang**
This PR removes the automatic transformation of URLs to lowercase across various modules. Many URLs are case-sensitive, and forcing them to lowercase may result in broken links or incorrect processing

Closes https://github.com/projectdiscovery/urlfinder/issues/35

### CODE REVIEW

**Review by ehsandeep — CHANGES_REQUESTED** [review]
Lint error

File: pkg/extractor/regex_extractor.go [hunk 1]
```diff
@@ -22,8 +21,6 @@ func NewRegexUrlExtractor() (*RegexUrlExtractor, error) {
 // Extract implements the UrlExtractor interface, using the regex to find urls in the given text.
 func (re *RegexUrlExtractor) Extract(text string) []string {
 	matches := re.extractor.FindAllString(text, -1)
-	for i, match := range matches {
-		matches[i] = strings.ToLower(match)
-	}
+	// The copy step is unnecessary here; you can directly return matches
```
Comment (dogancanbakir) [inline pkg/extractor/regex_extractor.go:24]: Let's do
`return matches` OR
 `return re.extractor.FindAllString(text, -1)` and remove the unnecessary comment.

**Review by dogancanbakir — APPROVED** [review]
LGTM -left a comment for you.

File: pkg/extractor/regex_extractor.go (hunk 1 above)
Comment (mujtabachang) [reply pkg/extractor/regex_extractor.go]: Comment is removed

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**